ddtrace/internal/ci_visibility/_api_client.py

```diff
@@ -366,102 +433,107 @@ def _do_skippable_request(self, timeout: float) -> t.Optional[_SkippableResponse
-                error_type = ERROR_TYPES.BAD_JSON
-                return None
+        covered_files: t.Optional[t.Dict[str, CoverageLines]] = None
 
-            return parsed
+        if skippable_response is None:
+            # We did not fetch any data, but telemetry has already been recorded, and a warning has been logged
+            return None
 
-        except Exception:
-            log.warning("Error retrieving skippable test data, no tests will be skipped", exc_info=True)
-            error_type = ERROR_TYPES.UNKNOWN
+        meta = skippable_response.get("meta")
+        if meta is None:
+            log.debug("SKippable tests response did not contain metadata field, no tests will be skipped")
+            record_api_request_error(metric_names.error, ERROR_TYPES.BAD_JSON)
             return None
 
-        finally:
-            record_itr_skippable_request(
-                sw.elapsed() * 1000,
-                response_bytes,
-                TEST if self._itr_skipping_level == ITR_SKIPPING_LEVEL.TEST else SUITE,
-                error=error_type,
-            )
+        correlation_id = meta.get("correlation_id")
+        if correlation_id is None:
+            log.debug("Skippable tests response missing correlation_id")
+        else:
+            log.debug("Skippable tests response correlation_id: %s", correlation_id)
 
-    def fetch_skippable_items(
-        self, timeout: t.Optional[float] = None, ignore_test_parameters: bool = False
-    ) -> t.Optional[ITRData]:
-        if timeout is None:
-            timeout = DEFAULT_ITR_SKIPPABLE_TIMEOUT
+        covered_files_data = meta.get("coverage")
+        if covered_files_data is not None:
+            covered_files = _parse_covered_files(covered_files_data)
 
-        error_type: t.Optional[ERROR_TYPES] = None
+        items_to_skip_data = skippable_response.get("data")
+        if items_to_skip_data is None:
+            log.warning("Skippable tests request missing data, no tests will be skipped")
+            record_api_request_error(metric_names.error, ERROR_TYPES.BAD_JSON)
+            return None
 
-        skippable_response = self._do_skippable_request(timeout)
+        if self._itr_skipping_level == ITR_SKIPPING_LEVEL.TEST:
+            items_to_skip = _parse_skippable_tests(items_to_skip_data, ignore_test_parameters)
+        else:
+            items_to_skip = _parse_skippable_suites(items_to_skip_data)
+        return ITRData(
+            correlation_id=correlation_id,
+            covered_files=covered_files,
+            skippable_items=items_to_skip,
+        )
 
-        covered_files: t.Optional[t.Dict[str, CoverageLines]] = None
+    def fetch_unique_tests(self) -> t.Optional[t.Set[InternalTestId]]:
+        metric_names = APIRequestMetricNames(
+            EARLY_FLAKE_DETECTION_TELEMETRY.REQUEST,
+            EARLY_FLAKE_DETECTION_TELEMETRY.REQUEST_MS,
+            EARLY_FLAKE_DETECTION_TELEMETRY.RESPONSE_BYTES,
```

Comment:
This is the only one that has `RESPONSE_BYTES` in the name, are the other metrics used as the `response_bytes` parameter also number of bytes? Just to make sure.

Reply:
They are but you made me (re)realize that we don't record the settings response bytes, so I have to make an exception for it. I shall refactor a bit.